tap_csv/client.py

```diff
@@ -67,10 +67,11 @@ def get_file_paths(self) -> list:
         file_paths = []
         if os.path.isdir(file_path):
             clean_file_path = os.path.normpath(file_path) + os.sep
-            for filename in os.listdir(clean_file_path):
-                file_path = clean_file_path + filename
-                if self.is_valid_filename(file_path):
-                    file_paths.append(file_path)
+            for dirpath, _, filenames in os.walk(clean_file_path):
```

Comment:
@TyShkan the sonarcloud scans think this method is too complex now. What if we refactor this for loop out to its own private method `_get_recursive_file_paths` or similar that returns a list like

```
        if os.path.isdir(file_path):
            clean_file_path = os.path.normpath(file_path) + os.sep
            file_paths = self._get_recursive_file_paths(clean_file_path)
```